desktop/app/src/init.tsx

```diff
@@ -213,9 +213,20 @@ function init() {
   sideEffect(
     store,
     {name: 'loadTheme', fireImmediately: false, throttleMs: 500},
-    (state) => ({
-      dark: state.settingsState.darkMode,
-    }),
+    (state) => {
```

Comment:
Mac switches dynamically between light and dark based on time (if set up), so I guess we have to set some system listener for that to reflect that, in case the mode is auto?

Reply:
The code 2 lines down is using the Electron API to tap into this, but is it not listening correctly to changes to update dynamically when the system switches?

`let shouldUseDarkMode = remote.systemPreferences.isDarkMode();`

Reply:
Maybe we can subscribe the `AppleInterfaceThemeChangedNotification` event and refresh the UI when the system theme changes? 

But my concern is that the UI refresh could be interruptive, is there a way to update the application theme without a refresh?

Reply:
I think the process being interruptive is fine, I see more applications doing that. I think the clue is to listen to the events exposed here https://www.electronjs.org/docs/api/native-theme in the flipper entry file, roughly following https://www.electronjs.org/docs/tutorial/dark-mode (which is partially already implemented). Updating the setting in the Redux store when the updated event happens, should apply the change automatically without needing a restart.

Reply:
I have replaced the `systemPreferences.isDarkMode()` with `nativeTheme.shouldUseDarkColors`. The application can now automatically react to the system theme changes without the need of UI refresh :)

@passy could you please review my changes again? Thanks.